rrtmgp/kernels-openacc/mo_gas_optics_kernels.F90

```diff
@@ -67,14 +67,14 @@ subroutine interpolation( &
     ! local indexes
     integer :: icol, ilay, iflav, igases(2), itropo, itemp
 
-    !$acc enter data copyin(flavor,press_ref_log,temp_ref,vmr_ref,play,tlay,col_gas)
+    !$acc data copyin(flavor,press_ref_log,temp_ref,vmr_ref,play,tlay,col_gas) &
+    !$acc      copyout(jtemp,jpress,tropo,jeta,col_mix,fmajor,fminor) &
+    !$acc      create(ftemp,fpress)
```

Comment:
Why the change from unstructured to structured data region? It is it trying to get around a compiler bug?

Reply:
Actually, at a later point in time I'd like to change to structured data sections whenever possible. It's simpler, less error-prone and potentially more efficient.
In this particular place structured section helps reduce the code a bit. Variables with indirect indexing (like `temp_ref` or `vmr_ref`) should better be explicitly listed as `present` in the kernels. Compiler may not be able to identify which part of the variable it has to apply implicit data sections to, and this may lead to a bug. I latest NVHPC version work fine in this case, but I want to make sure that's the case.
Then I just wanted to combine the said annotation in the kernel with the allocation/deallocation, which becomes exactly the structured data section.

Reply:
I agree that structure OpenMP data regions are general much less error prone. I think it would be good to change the OpenMP offload to minimize divergence. Could it get done in this pull request?

Reply:
Sure, I'll update it after the weekend